Client/Extensions/AsyncAwait.swift

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+
+extension Sequence {
+    func asyncForEach(_ operation: (Element) async throws -> Void) async rethrows {
+        for element in self {
+            try await operation(element)
+        }
+    }
+    
+    func asyncConcurrentForEach(_ operation: @escaping (Element) async throws -> Void) async rethrows {
+        try await withThrowingTaskGroup(of: Void.self) { group in
+            for element in self {
+                group.addTask { try await operation(element) }
+            }
+            
+            try await group.waitForAll()
```

Comment:
just fyi you don't need to call this, `await withThrowingTaskGroup` doesn't return until all tasks complete.